src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -263,17 +263,64 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
     mOperationalAdvertisingParams = params;
     // TODO: There may be multilple device/fabrid ids after multi-admin.
 
-    ReturnErrorOnFailure(SetupHostname(params.GetMac()));
+    // According to spec CRI and CRA intervals should not exceed 1 hour (3600000 ms).
+    // TODO: That value should be defined in the ReliableMessageProtocolConfig.h,
+    // but for now it is not possible to access it from src/lib/mdns. It should be
+    // refactored after creating common DNS-SD layer.
+    constexpr uint32_t kMaxCRMPRetryInterval = 3600000;
+    TextEntry crmpRetryIntervalEntries[2];
+    size_t textEntrySize = 0;
+    char crmpRetryIntervalIdleBuf[8];
+    char crmpRetryIntervalActiveBuf[8];
```

Comment:
If we're going to ignore the returned values from snprintf, I think we need comments on why these buffers are definitely big enough, a defined constant for the lengths, and if possible a compile-time verification of all this.

This code will work, but assumes the buffers are big enough because `kMaxCRMPRetryInterval` in decimal can fit into 7 + 1 chars.  If `kMaxCRMPRetryInterval` changes and the buffers aren't resized, this can break.  A comment to this effect seems like the bare minimum.   Likewise, changing the printf format string can also break things.  For this reason, the format string should probably also be a constant defined here with everything else.

Reply:
Yes, I have to agree with this. All of this values should be probably defined in the ReliableMessageProtocolConfig.h, however considering current structure messaging module cannot be directly accessed from the src/lib/mdns and all values should be passed through the AdvertisingParameters, what would be kind of weird. I think we are planning to refactor those files soon and make it accessible easier and that's why I must admit that I did it a little bit sloppy. I tried to address your remarks and make the code at least less error prone.

Reply:
Thank you.  Looks good to me.